backend/src/apiserver/server/experiment_server.go

```diff
@@ -122,7 +133,7 @@ func (s *ExperimentServer) canAccessExperiment(ctx context.Context, experimentID
 		return util.Wrap(err, "Failed to authorize with the experiment ID.")
 	}
 	if len(namespace) == 0 {
-		return util.NewInternalServerError(errors.New("There is no namespace found"), "There is no namespace found")
+		return util.NewInternalServerError(errors.New("Empty namespace"), "The experiment doesn't have a valid namespace.")
```

Comment:
nit: not a big issue, would it make sense to allow access to experiment with namespace="" blindly?
because these namespaces are coming from before multi user mode was enabled.

I think you can decide whatever makes sense to you.

Reply:
Initially, I allow listing experiments with empty namespace, but reject accessing experiment with empty namespace. The idea was to allow user see a list, but when they click on it, they get an error message. 
Now that we disallowed empty namespace during listing, the experiments with empty namespace is almost invisible. So I'm leaning towards keeping it consistent for both list and get.
Eventually, this shouldn't happen once backfill is done (either via a SQL script or by automated backfill from api-server).